.github/workflows/lint-pr.yml

```diff
@@ -0,0 +1,20 @@
+name: "Lint PR"
```

Comment:
I feel this name doesn't make it abundanly clear what this workflow would do. Maybe we should do something like
`Enforce semantic pull request title`